pubsub/google/cloud/pubsub_v1/publisher/batch/mp.py

```diff
@@ -0,0 +1,398 @@
+        and then calls commit unless the batch has already been committed.
+        """
+        # Note: This thread blocks; it is up to the calling code to call it
+        # in a separate thread.
+        #
+        # Sleep for however long we should be waiting.
+        time.sleep(self.settings.max_latency)
+
+        # If, in the intervening period, the batch started to be committed,
+        # then no-op at this point.
+        if self.status != self.Status.ACCEPTING_MESSAGES:
+            return
+
+        # Commit.
+        return self._commit()
+
+    def publish(self, data, **attrs):
+        """Publish a single message.
+
+        .. note::
+            Messages in Pub/Sub are blobs of bytes. They are *binary* data,
+            not text. You must send data as a bytestring
+            (``bytes`` in Python 3; ``str`` in Python 2), and this library
+            will raise an exception if you send a text string.
+
+            The reason that this is so important (and why we do not try to
+            coerce for you) is because Pub/Sub is also platform independent
+            and there is no way to know how to decode messages properly on
+            the other side; therefore, encoding and decoding is a required
+            exercise for the developer.
+
+        Add the given message to this object; this will cause it to be
+        published once the batch either has enough messages or a sufficient
+        period of time has elapsed.
+
+        Args:
+            data (bytes): A bytestring representing the message body. This
+                must be a bytestring (a text string will raise TypeError).
+            attrs (Mapping[str, str]): A dictionary of attributes to be
+                sent as metadata. (These may be text strings or byte strings.)
+
+        Raises:
+            TypeError: If the ``data`` sent is not a bytestring, or if the
+                ``attrs`` are not either a ``str`` or ``bytes``.
+
+        Returns:
+            ~.pubsub_v1.publisher.batch.mp.Future: An object conforming to the
+                :class:`concurrent.futures.Future` interface.
+        """
+        # Sanity check: Is the data being sent as a bytestring?
+        # If it is literally anything else, complain loudly about it.
+        if not isinstance(data, six.binary_type):
+            raise TypeError('Data being published to Pub/Sub must be sent '
+                            'as a bytestring.')
+
+        # Coerce all attributes to text strings.
+        for k, v in copy.copy(attrs).items():
+            if isinstance(data, six.text_type):
+                continue
+            if isinstance(data, six.binary_type):
+                attrs[k] = v.decode('utf-8')
+                continue
+            raise TypeError('All attributes being published to Pub/Sub must '
+                            'be sent as text strings.')
+
+        # Store the actual message in the batch's message queue.
+        self._shared.messages.append(
```

Comment:
It does, in fact, work.

The point that is being made in the docs is that if you modified _an item within the list_, it would not be propagated correctly.

Perhaps an example to make it clearer:

```python
obj = [1, 2, 3]
my_list = manager.list()
my_list.append(obj)  # This gets propagated.
obj.append(4)  # This does not get propagated.
```

I did not understand this either until I dug into it for Pub/Sub.